### ISSUE TITLE
why does firmware flashing sketch not use Wire?

### ISSUE THREAD

**Issue by tlyu** [keyboardio/avr_keyscanner#19]
What the reason was for using (a snapshot of?) the `twi.c` internals from the Wire library instead of Wire? I'm suspecting some sort of incompatibility, but I'm not finding many clues in the history. Will it work on the GD32-based (?) Model 100 without porting to Wire? The Kaleidoscope hardware support for the Model 100 seems to use Wire to talk to the ATtinys, from what I can tell.

**Comment by obra** [comment]
At the time, I believe to be primary reason for the cut down implementation was space. Wire includes both master and slave implementations and isn’t able to optimize them out. On the Model 01, we were very very flash constrained. It’s been a while and there may have been some perf implications too, but space was the primary concern. 

For non avr devices, no, that implementation won’t work and since we’re not nearly as space constrained, just using Wire is a reasonable path to a first implementation.

**Comment by tlyu** [comment]
I think the firmware flashing sketch, even on the Model 01, is much less space constrained, so it might be safe to switch that back to using Wire?

**Comment by obra** [comment]
Yep. At one point we were hoping to bundle it into keyboard firmware, but that never happened on the Model 01
